Source/Mosa.Compiler.x64/Intrinsic/IRQs.cs

```diff
@@ -42,7 +42,7 @@ private static void InsertIRQ(int irq, Context context, Transform transform)
 
 		var interrupt = Operand.CreateLabel(method, transform.Is32BitPlatform);
 
-		var esp = Operand.CreateCPURegister32(CPURegister.RSP);
+		var esp = transform.PhysicalRegisters.Allocate32(CPURegister.RSP);
```

Comment:
wrong register size used, 32bit register on 64bit architecture
should be rsp instead of esp and should use Allocate64 instead of Allocate32

Reply:
Nice catch!